src/app/InteractionModelEngine.cpp

```diff
@@ -122,6 +132,8 @@ void InteractionModelEngine::Shutdown()
     mEventPathPool.ReleaseAll();
     mDataVersionFilterPool.ReleaseAll();
     mpExchangeMgr->UnregisterUnsolicitedMessageHandlerForProtocol(Protocols::InteractionModel::Id);
+
```

Comment:
should we also set `mpExchangeMgr = nullptr` ? and if we do, is mState the same as mpExchangeMgr being nullptr or not?

Reply:
I agree: we should null out `mpExchangeMgr` here, and we should just null-check it at the start of shutdown.

Reply:
I did update the code to not use any `State` enum and just null check for both `mpExchangeContext` and `mpFabricTable`